File: app/src/main/java/com/amperas17/mangaedenapp/api/responseprovider/ChapterResponseProvider.java
package com.amperas17.mangaedenapp.api.responseprovider;

import com.amperas17.mangaedenapp.MangaEdenApp;
import com.amperas17.mangaedenapp.model.page.PageListResponse;
import com.amperas17.mangaedenapp.utils.Caller;

import retrofit2.Call;


public class ChapterResponseProvider extends BaseResponseProvider<PageListResponse, String> {
    public ChapterResponseProvider(Caller<PageListResponse> caller) {
        super(caller);
    }

    @Override
    public Call<PageListResponse> initCall(String... args) {
        return MangaEdenApp.getMangaApi().getPageList(args[0]);
    }
}

File: app/src/main/java/com/amperas17/mangaedenapp/api/responseprovider/ChapterResponseProvider.kt
package com.amperas17.mangaedenapp.api.responseprovider

import com.amperas17.mangaedenapp.MangaEdenApp
import com.amperas17.mangaedenapp.model.page.PageListResponse
import com.amperas17.mangaedenapp.utils.Caller

import retrofit2.Call


class ChapterResponseProvider(caller: Caller<PageListResponse>) : BaseResponseProvider<PageListResponse, String>(caller) {

    override fun initCall(vararg args: String): Call<PageListResponse> {
        return MangaEdenApp.mangaApi.getPageList(args[0])
    }
}